### DISCUSSION TITLE
The "ikm" argument must be of type string or an instance of SecretKeyObject, ArrayBuffer, TypedArray, DataView, or Buffer. Received undefined

### DISCUSSION THREAD

**Discussion by ev4ndropc** [nextauthjs/next-auth#3256] [Help]
### Question 💬

Hi, I'm getting this error and I have no idea why!
I've already searched the entire documentation for next-auth, on google and I can't find anything related to it. can you help me?

OBS: I can't post my code here because it breaks the formatting, I'll leave the file link in my repository

`https://github.com/ev4ndropc/Rocket_Controller/blob/main/src/pages/api/auth/%5B...nextauth%5D.js`

### How to reproduce ☕️

The error happens every time I try to login
![Screenshot_1](https://user-images.githubusercontent.com/34420396/142879903-d2990da9-0a9b-4509-82c7-9b4d428ff188.png)

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
I don't see a top level secret defined.

https://next-auth.js.org/getting-started/upgrade-v4#missing-secret

Also, where did you read about `expiresIn` in the jwt config? It's not something I'm familiar with.

**Comment by ev4ndropc** [comment]
The strange thing is that in my house where I use windows 10 and on vercel it works normally, at my work where I use windows 11 it doesn't work

**Reply by balazsorban44** [reply]
could be some corporate proxy setting that messes with cookies maybe?

**Reply by brennongs** [reply]
I'm also getting this error, but everyone else on my team (on the same network) is working fine. We have duplicate environment variables etc., and there are a mix of M1 macs and Intel based macs in the mix, so I'm not really sure what could be different.

**Comment by Baraka-Mulumia** [comment]
In development hardcode your JWT secret and secret for next-auth.

**Comment by jaxomlotus** [comment]
I'm running into this issue as well. My JWT and next-auth secret are hardcoded, and it doesn't help.

**Reply by Baraka-Mulumia** [reply]
Is both the JWT secret and the secret a string of 32 characters?

**Reply by jaxomlotus** [reply]
Not currently, but when I tried setting it to be one generated using this it didn't make a difference: 

`openssl rand -base64 32`

I didn't see it being 32 characters as a requirement in the documentation. But if it's a strict requirement for the secret but I will try that again

**Reply by Baraka-Mulumia** [reply]
It is not a strict requirement but as for me thats what solved the problem in development mode. Was thinking might be the same issue as you. But keep trying  hopefully  a permanent  solution to this will  come along soon

**Reply by jaxomlotus** [reply]
ok, I'm actually noticing that `openssl rand -base64 32` generates 44 characters. maybe that's part of the problem? i'll try truncating it to 32. Thanks for the help

**Reply by Baraka-Mulumia** [reply]
Sample "64cb5a072f7ba2d935224006bb56bfa8";

**Reply by jaxomlotus** [reply]
Unfortunately this didn't help. Even trying it with your string just resulted in ` TypeError [ERR_INVALID_ARG_TYPE]: The "ikm" argument must be of type string or an instance of SecretKeyObject, ArrayBuffer, TypedArray, DataView, or Buffer. Received undefined  `

**Reply by Baraka-Mulumia** [reply]
Could you share your  _next auth_ file if you don't mind.

**Comment by yashdilip** [comment]
I had the similar issue, the app was working fine in my local and not in Vercel deployed instance.

It got resolved by adding SECRET (for me it is 'JWT_SECRET') environment variable in Vercel environment variables. I was using this JWT_SECRET in my [...nextAuth].js file

`
jwt: {
    secret: process.env.JWT_SECRET,
  },`

**Comment by dingdangdog** [comment]
Today, the same problem occurs when I use [sidebase/nuxt-auth](https://github.com/sidebase/nuxt-auth) And `jsonwebtoken` to build porject. 

When I use `Credentials` provider to request `signIn` , I get an error message saying:

```
[nuxt] [request error] [unhandled] [500] "ikm"" must be an instance of Uint8Array or a string
```

**And When I run `npm uninstall jsonwebtoken`  to remove the dependency, this error disappears.**

I know `nuxt-auth` is based on `next-auth` developed, so I write it here, hope it helps.

> By the way, can't `nuxt-auth/next-auth` and `jsonwebtoken` be used at the same time?
